components/core/tests/test-EncodedVariableInterpreter.cpp

```diff
@@ -376,6 +376,43 @@ TEST_CASE("EncodedVariableInterpreter", "[EncodedVariableInterpreter]") {
         ));
     }
 
+    SECTION("Test multiple metching values") {
+        char const cVarDictPath[] = "var.dict";
+        char const cVarSegmentIndexPath[] = "var.segindex";
+        vector<string> var_strs = {"python2.7.3", "Python2.7.3", "PyThOn2.7.3", "PYTHON2.7.3"};
+        clp::VariableDictionaryWriter var_dict_writer;
+        var_dict_writer.open(cVarDictPath, cVarSegmentIndexPath, cVariableDictionaryIdMax);
+        vector<encoded_variable_t> encoded_vars;
+        vector<clp::variable_dictionary_id_t> var_ids;
+        clp::LogTypeDictionaryEntry logtype_dict_entry;
+        for (auto var_str : var_strs) {
+            EncodedVariableInterpreter::encode_and_add_to_dictionary(
```

Comment:
Instead of directly using the var_strs as msg argument for encode_and_add_to_dictionary, can we construct a message with a dummy log type?

I guess what you can do can be similar to

```
string const msg_template  = "here is a string with a dictionary var: ".

for var_str in var_strs:
    auto const msg = msg_template + var_id
    ...
```